libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1835,16 +1829,17 @@ bool PbftManager::pushPbftBlock_(PeriodData &&period_data, std::vector<std::shar
     pbft_chain_->updatePbftChain(pbft_block_hash, null_anchor);
   }
 
-  // Advance pbft consensus period
-  advancePeriod();
-
   LOG(log_nf_) << "Pushed new PBFT block " << pbft_block_hash << " into chain. Period: " << pbft_period
                << ", round: " << getPbftRound();
 
-  finalize_(std::move(period_data), std::move(dag_blocks_order));
+  finalize_(std::move(period_data), std::move(dag_blocks_order), synchronous_processing);
 
   db_->savePbftMgrStatus(PbftMgrStatus::ExecutedBlock, true);
   executed_pbft_block_ = true;
+
+  // Advance pbft consensus period
+  advancePeriod();
```

Comment:
I guess better design would be to have all this in call back function of `finalize`

Reply:
any reason ?

Reply:
actually, it cannot be in callback function if we want to take advantage of asynchronous processing and continue advance period asap

Reply:
instead of synchronous execution, you should execute it in callback function and use async exe

Reply:
? please re review the code as I pushed some changes because I dont really understand how is what you suggest relevant to this code

Reply:
in other words I dont get it so you push commit with change - that will be faster